### ISSUE TITLE
Emacs uses CPU when this mode is enabled

### ISSUE THREAD

**Issue by oscarfv** [kiennq/emacs-mini-modeline#53]
First of all: very nice package, thank you!

I noticed that with this mode active Emacs uses ~1-2% of one CPU core when sitting idle. I tried setting `mini-modeline-update-interval` to a larger value (0.3), but that had no effect.

**Comment by oscarfv** [comment]
The problem comes from the timer. Currently I'm experimenting with an idle timer instead.

By the way, this code:

```
(when (timerp (cancel-timer mini-modeline--timer)))
```

probably should be:

```
(when (timerp mini-modeline--timer) (cancel-timer mini-modeline--timer))
```

**Comment by kiennq** [comment]
Nice. PR is welcome :D

**Comment by slotThe** [comment]
This is perhaps tangentially related, but I wonder why the `mini-modeline--timer` is always called with a repeat of 0.1 

``` emacs-lisp
(setq mini-modeline--timer (run-with-timer 0 0.1 #'mini-modeline-display))
```

and _then_ we are checking if we actually have the right update interval in `mini-modeline-display` 

``` emacs-lisp
when (or (memq arg '(force clear))
                          (mini-modeline--overduep mini-modeline--last-update
                                                   mini-modeline-update-interval))              
```

Wouldn't it make more sense to call 

``` emacs-lisp
(setq mini-modeline--timer (run-with-timer 0 mini-modeline-update-interval #'mini-modeline-display))
```

directly and get rid of the `mini-modeline--last-update` variable entirely (which, as far as I can tell, is not used anywhere else)?

**Comment by slotThe** [comment]
> @slotThe, when using an idle-timer, as in #57, I believe it makes more sense to do this kind of check, since the idle timer can fire kind of whenever. But then again, if `mini-modeline-update-interval` <= the timer interval, (both are 0.1 in by default in #57 it would probably mean that it should always update when it is called (through the timer).

Indeed, I think with an idle timer—which seems like the best solution anyways—it would make sense again
